pywbem/_logging.py

```diff
@@ -0,0 +1,407 @@
+    class is a singleton, there is only one set of data for a pywbem
+    instantiation.  It goal is to defined named loggers from data input but
+    it also records records about these named loggers in a dictionary
+
+    This is defined with only class level object and methods. This is an
+    easy way to create a singleton. However, at least in python 2.6, you
+    cannot make some magic methods work (ex.__getItem__)
+
+    There are two constructors for loggers:
+
+      PywbemLoggers.create_loggers - Creates one or more logger entries in
+      this class and also in the python logging class from an input string
+      with the format defined in the create_loggers method below.
+
+      PywbemLoggers.create_logger - Creates a single logger fromthe
+      separate pywbem logging attributes.
+
+    """
+    __metaclass__ = MetaPywbemLoggers
+
+    @classmethod
+    def reset(cls):
+        """Reset the logger dictionary. Used primarily in unittests"""
+        cls.loggers = {}
+
+    @classmethod
+    def create_loggers(cls, input_str, log_filename=None):
+        """
+        Create the pywbem loggers defined by the input string in the following
+        syntax and place the logger definitions in the class level dictionary
+        in this class.
+
+        Parameters:
+          input_str (:term:`string`) that specifies the logger definitions:
+            as follows:
+
+            ``log_specs := log_spec[,log_spec]``
+
+            ``log_spec := log_comp['=' [dest][":"[detail_lvl][":"[log_lvl]]]]``
+
+            where:
+                ``log_comp`` must be one of strings in the LOG_COMPONENTS list.
+                ``log_lvl`` must be one of strings in the LOG_LEVELS list.
+                ``detail_lvl`` must be one of strings in the
+                    LOG_DETAIL_LEVELS list
+
+                ``dest`` must be one of strings in the LOG_DESTINATIONS list
+
+          log_filename (:term:`string`)
+            Optional string that defines the filename for output of logs
+            if the dest type is `file`
+
+        Exceptions:
+          ValueError - Generated if the syntax of the input string is invalid
+          or any of the components is not one of allowed strings.
+
+        Examples:
+            ``ops=stderr:min,debug``    # set cim operations logger
+            ``http=file::debug``        # set http logger to send to file
+            ``all=file:all:debug``      # Set all named loggers to
+        """
+        results = cls._parse_log_specs(input_str)
+        for name, value in six.iteritems(results):
+            cls.create_logger(name, log_dest=value[0],
+                              log_detail_level=value[1],
+                              log_level=value[2],
+                              log_filename=log_filename)
+
+    @classmethod
+    def create_logger(cls, log_component, log_dest='stderr',
+                      log_filename=None,
+                      log_detail_level='all', log_level='debug'):
```

Comment:
The name of the `create_logger()` method should be changed to e.g. `get_logger()` because it really does what Python `logging.getLogger()` does: Return the existing logger if it exists, or create a new logger otherwise.